src/sentry/workflow_engine/handlers/detector/stateful.py

```diff
@@ -316,16 +470,16 @@ def evaluate_group_key_value(
         value: DataPacketEvaluationType,
         state_data: DetectorStateData,
         dedupe_value: int,
+        data_packet: DataPacket[DataPacketType],
     ) -> DetectorEvaluationResult | None:
         """
         Evaluates a value associated with a given `group_key` and returns a `DetectorEvaluationResult` with the results
         and any state changes that need to be made.
 
         Checks that we haven't already processed this datapacket for this group_key, and skips evaluation if we have.
         """
+        # TODO - compose this method using the helpers in the base class.
```

Comment:
I think there's still a bit more we can decompose out of this class, but this was starting to get a bit untenable for review.